soh/src/code/z_player_lib.c

```diff
@@ -1068,13 +1094,30 @@ s32 func_800902F0(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s
             *dList = sArmOutDLs[gSaveContext.linkAge];
         } else if (limbIndex == PLAYER_LIMB_L_HAND) {
             *dList = sHandOutDLs[gSaveContext.linkAge];
+
+            if(CVarGetInteger("gBowSlingShotAmmoFix", 0)){
+                if (Player_HoldsSlingshot(this)) {
+                    *dList = NULL;
+                } else if (LINK_IS_ADULT && (Player_HoldsBow(this) || Player_HoldsSlingshot(this))) {
+                    *dList = gLinkAdultRightHandOutNearDL;
+                }
+            }
```

Comment:
couple things here, first, it's not clear why you have
```c
} else if (LINK_IS_ADULT && (Player_HoldsBow(this) || Player_HoldsSlingshot(this))) {
```
when `Player_HoldsSlingshot(this)` will never be true because of the `else`

second, instead of overriding the already set dlist it probably makes sense to just conditionally set the index

```suggestion
            s32 handOutDlIndex = gSaveContext.linkAge;
            if (CVarGetInteger("gBowSlingShotAmmoFix", 0)) {
                 if (Player_HoldsSlingshot(this)) {
                    handOutDlIndex = 1;
                } else if (LINK_IS_ADULT && Player_HoldsBow(this)) {
                    handOutDlIndex = 0;
                }
            }

            *dList = sHandOutDLs[handOutDlIndex];
```

Reply:
I think this block is the result of going back and forth while testing and not cleaning up after. All this should be doing is removing Adult Link's giant left hand when using the slingshot.

https://user-images.githubusercontent.com/22593559/218890514-00835893-b819-41a9-8902-b73b48d872ab.mp4

If we want to keep the hand this entire block can go away, otherwise it can be simplified:
```suggestion
            s32 handOutDlIndex = gSaveContext.linkAge;
            if (CVarGetInteger("gBowSlingShotAmmoFix", 0) && LINK_IS_ADULT && Player_HoldsSlingshot(this)) {
                handOutDlIndex = 1;
            }
            *dList = sHandOutDLs[handOutDlIndex];
```

Thoughts?

Reply:
i'm just saying what you had was odd because
```c
if (Player_HoldsSlingshot(this)) {
  //DO SLINGSHOT STUFF
} else if (LINK_IS_ADULT && (Player_HoldsBow(this) || Player_HoldsSlingshot(this)) {
  //WE WILL NEVER GET HERE WITH A SLINGSHOT BECAUSE OF THE ELSE
  //SO WHY DO WE HAVE "|| Player_HoldsSlingshot(this)" IN HERE
}
```

Reply:
i'm happy with the simplified version that removes the giant hand

Reply:
Yeah I know what you meant with the logic, while fixing it I realized I was making something simple overly complicated.

Reply:
i got thrown off by the mention of showing the hands and didn't make the connection that you new suggested simplified code wasn't showing it

sorry for the all caps in the code comments there, the video response really threw me for a loop